### PR TITLE
[release/v2.20] Fix vSphere integration and e2e tests

### PR DESCRIPTION
**Pull request by ahmedwaleedmalik**
**What this PR does / why we need it**:
Manual backport of https://github.com/kubermatic/kubermatic/pull/10918

**Which issue(s) this PR fixes**:

Fixes #

**What type of PR is this?**

/kind bug
/kind failing-test

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
NONE
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

File: hack/ci/testdata/seed.yaml [hunk 1]
```diff
@@ -71,12 +71,14 @@ spec:
           datacenter: Hamburg
           datastore: alpha1
           endpoint: https://vcenter.dc.k8c.io
-          rootPath: /Hamburg/vm/Kubermatic-dev
+          ipv6Enabled: true
```
Comment (embik) [inline hack/ci/testdata/seed.yaml]: `ipv6Enabled` needs to be removed.

**Review by embik — CHANGES_REQUESTED** [review]

File: hack/ci/testdata/seed.yaml (hunk 1 above)
Comment (ahmedwaleedmalik) [reply hack/ci/testdata/seed.yaml]: Updated, good catch 😁

**Review by embik — APPROVED** [review]

File: hack/ci/setup-mla.sh [hunk 2]
```diff
@@ -80,4 +80,4 @@ kubectl port-forward --address 0.0.0.0 -n mla svc/cortex-ruler 3002:8080 > /dev/
 kubectl port-forward --address 0.0.0.0 -n mla svc/loki-distributed-ruler 3003:3100 > /dev/null &
 echodate "Finished exposing components"
 
-echodate "MLA is ready."
+echodate "MLA is ready."
```
Comment (embik) [inline hack/ci/setup-mla.sh]: looks like `shfmt` isn't happy about the newline removal here.

**Review by embik — CHANGES_REQUESTED** [review]

**Comment by ahmedwaleedmalik** [comment]
/retest

Seems like a flake

**Comment by ahmedwaleedmalik** [comment]
@LittleFox94 Yes I'm investigating and working on fixing those failures. We don't have a 2.20 release planned anytime soon so I hope it's fine that your PR stick arounds for a little longer.

File: pkg/test/e2e/ccm-migration/ccm_migration_test.go [hunk 3]
```diff
@@ -42,7 +42,7 @@ import (
 
 const (
 	userClusterPollInterval = 5 * time.Second
-	customTestTimeout       = 10 * time.Minute
+	customTestTimeout       = 30 * time.Minute
```
Comment (ahmedwaleedmalik) [inline pkg/test/e2e/ccm-migration/ccm_migration_test.go:45]: Tried with 15 minutes, this is what we currently have in master, but it still timeouts although the migration worked fine.

**Pull request merged**